pylint/extensions/comment_ending.py

```diff
@@ -0,0 +1,56 @@
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IRawChecker
+
+
+def is_line_commented(line):
+    """ Checks if a `# symbol that is not part of a string was found in line"""
+
+    comment_idx = line.find(b"#")
+    if comment_idx == -1:
+        return False
+    if comment_part_of_string(line, comment_idx):
+        return is_line_commented(line[:comment_idx] + line[comment_idx + 1 :])
+    return True
+
+
+def comment_part_of_string(line, comment_idx):
+    """ checks if the symbol at comment_idx is part of a string """
+
+    if (
+        line[:comment_idx].count(b"'") % 2 == 1
+        and line[comment_idx:].count(b"'") % 2 == 1
+    ) or (
+        line[:comment_idx].count(b'"') % 2 == 1
+        and line[comment_idx:].count(b'"') % 2 == 1
+    ):
+        return True
+    return False
+
+
+class CommentChecker(BaseChecker):
+    __implements__ = IRawChecker
+
+    name = "refactoring"
+    msgs = {
+        "R2044": (
+            "Unnecessary comment at end of line",
```

Comment:
I think an empty comment with no code at all on the line is also a problem.

Reply:
Yeah, my checker already covers that. It's a good case and I will add it to the test case as you recommended and update the unit-test accordingly